### PR TITLE
Content/js movie demo

### PR DESCRIPTION
**Pull request by whatSocks**
[ ] wait for updated movies

### CODE REVIEW

File: modules/ROOT/pages/js-movie-app.adoc [hunk 1]
```diff
@@ -122,7 +122,8 @@ Now you’re looking for a demo app or template to start putting those skills in
 
 If you haven't already, clone the https://github.com/neo4j-examples/neo4j-movies-template[Neo4j Movie Template] onto your machine.
 This tutorial post will walk you through rating a movie on a sample movie rating application, from the initial setup to viewing the list of movies you’ve rated.
-The following video will walk briefly walk you through the various features of the movie app. 
+The following video will walk briefly walk you through the various features of the movie app.
+Although the title of the video is "Python App Walkthrough" the functionality of the JavaScript version should be identical.
```
Comment (fbiville) [inline modules/ROOT/pages/js-movie-app.adoc]: The title of the below video is "Movie App Walkthrough" for me. Maybe remove the warning here?

File: modules/ROOT/pages/js-movie-app.adoc [hunk 2]
```diff
@@ -165,25 +166,25 @@ You can find the front-end code in the `web` directory.
 
 === Setup
 
+<!--
```
Comment (fbiville) [inline modules/ROOT/pages/js-movie-app.adoc]: remove instead of commenting out?

File: modules/ROOT/pages/js-movie-app.adoc [hunk 3]
```diff
@@ -23,9 +23,9 @@ You should have cloned https://github.com/neo4j-examples/neo4j-movies-template[N
 This course provides an overview of what you need to know to write an Express application using a Neo4j database. 
 To keep concepts simple, the application is an IMDB clone with basic account authentication and movie recommendation functionality. 
 
-++++
+<!-- ++++
```
Comment (fbiville) [inline modules/ROOT/pages/js-movie-app.adoc]: remove instead of commenting out?

File: modules/ROOT/pages/js-movie-app.adoc [hunk 4]
```diff
@@ -235,15 +237,42 @@ Click on a cast or crew member to see that person’s profile, which includes bi
...
+----
+api.get("/genres", routes.genres.list);
+---- 
+
+The information from the call would be routed by `api/routes/genres.js` in and out of the the Genre model in `api/models/genres.js`, sending the response, an array of `Genres`, back to the user. 
+
+.api/models/genres.js
+[source,javascript,role=nocopy]
+----
+var _ = require('lodash');
```
Comment (fbiville) [inline modules/ROOT/pages/js-movie-app.adoc]: nit: all `var`s could be replaced with `const`s

File: modules/ROOT/pages/js-movie-app.adoc [hunk 5]
```diff
@@ -305,14 +334,35 @@ curl -X POST --header 'Content-Type: application/json' --header 'Accept: applica
...
 ----
-// TK User Registration Implementation
+api.post("/register", routes.users.register);
 ----
 
+.api/routes/users.js
+[source,javascript,role=nocopy]
+----
+exports.register = function (req, res, next) {
+  var username = _.get(req.body, 'username');
```
Comment (fbiville) [inline modules/ROOT/pages/js-movie-app.adoc]: same nit as above

File: modules/ROOT/pages/js-movie-app.adoc [hunk 6]
```diff
@@ -398,10 +448,62 @@ curl -X POST --header 'Content-Type: application/json' --header 'Accept: applica
...
+The information is routed through `api/routes/movies.js`:
+
+.api/routes/movies.js
 [source,javascript,role=nocopy]
 ----
-// TK JS Rate A Movie
+exports.rateMovie = function (req, res, next) {
+  loginRequired(req, res, () => {
+    var rating = Number(_.get(req.body, 'rating'));
+    if (isNaN(rating) || rating < 0 || rating >= 6) {
```
Comment (fbiville) [inline modules/ROOT/pages/js-movie-app.adoc]: nit: `Number.isNaN` is known to be a bit less surprising with some edge cases than the global `isNaN`

**Review by fbiville — CHANGES_REQUESTED** [review]

**Comment by whatSocks** [comment]
@fbiville the commenting out doesn't work - I'm updating the videos before merging

File: modules/ROOT/pages/js-movie-app.adoc (hunk 1 above)
Comment (whatSocks) [reply modules/ROOT/pages/js-movie-app.adoc]: waiting on movies to be updated

File: modules/ROOT/pages/js-movie-app.adoc (hunk 2 above)
Comment (whatSocks) [reply modules/ROOT/pages/js-movie-app.adoc]: waiting on movies to be updated

File: modules/ROOT/pages/js-movie-app.adoc (hunk 3 above)
Comment (whatSocks) [reply modules/ROOT/pages/js-movie-app.adoc]: waiting on movies to be updated

File: modules/ROOT/pages/js-movie-app.adoc (hunk 4 above)
Comment (whatSocks) [reply modules/ROOT/pages/js-movie-app.adoc]: done

File: modules/ROOT/pages/js-movie-app.adoc (hunk 5 above)
Comment (whatSocks) [reply modules/ROOT/pages/js-movie-app.adoc]: done

**Comment by fbiville** [comment]
@whatSocks is it ready for review?

**Review by fbiville — APPROVED** [review]

**Pull request merged**